docs/Setting-up-monorepo.md

```diff
@@ -0,0 +1,356 @@
+
+## Prerequisites
+
+Before you start, ensure you have the following installed on your machine:
+
+- [Node.js](https://nodejs.org/en) (version 18 or later)
+- [Yarn](https://classic.yarnpkg.com/en/)
+
+## Use case
+
+In a monorepo setup, multiple apps often share common functionality. For example, a mobile (React Native) and a web app (React). This tutorial demonstrates how to set up and use shared utilities within a monorepo. For instance, if you have several apps that need to validate user inputs like email addresses, passwords, or text fields, you can create a single validation utility that can be reused across all your apps.
+
+By centralizing these utilities, you reduce code duplication and simplify maintenance, ensuring that any updates or bug fixes are immediately available to all your apps.
+
+In this setup, we’ll create a React Native app, a React app and a form-validator utility that includes functions to validate email addresses, passwords, and generic text fields. This utility will then be integrated to both apps.
+
+## Step 1: Setting up the monorepo
+
+First, follow the [Expo documentation on setting up monorepos](https://docs.expo.dev/guides/monorepos/) to initialize your own monorepo. This will include setting up your `packages/` and `apps/` directories and configuring Yarn workspaces.
+
+Here's a quick recap:
+
+1. Initialize the monorepo:
+
+```
+mkdir monorepo-example
+cd monorepo-example
+yarn init
+```
+
+2. Configure workspaces in `package.json`:
+
+```
+{
+  "private": true,
+  "workspaces": {
+    "packages": [
+      "packages/*"
+    ],
+    "apps": [
+      "apps/*"
+    ]
+  }
+}
+```
+
+3. Create directory structure:
+
+```
+mkdir apps packages
+```
+
+## Step 2: Create mobile app using Ignite
+
+Ignite is a powerful tool for scaffolding React Native projects. In this step, we'll create your first app within the monorepo using Ignite.
+
+1. Install Ignite CLI (if you haven't already):
+
+```
+npm install -g ignite-cli
+```
+
+2. Generate a new app:
+   Navigate to the apps/ directory and run the following command to create a new app:
+
+```
+cd apps
+ignite new mobile
+```
+
+We suggest the following answers to the prompts:
+
+```
+📝 Do you want to use Expo?: Expo - Recommended for almost all apps [Default]
+📝 Which Expo workflow?: Expo Go - For simple apps that don't need custom native code [Default]
+📝 Do you want to initialize a git repository?: No
+📝 Remove demo code? We recommend leaving it in if it's your first time using Ignite: No
+📝 Which package manager do you want to use?: yarn
+📝 Do you want to install dependencies?: No
+```
+
+3. Open `metro.config.js` file
+
+```
+touch mobile/metro.config.js
+```
+
+4. Replace Metro configuration with the following content
```

Comment:
Instead of `replace`, show a diff instead. You can do a diff with three backticks & `diff`, like this:

```diff
diff --git a/old-metro.config.js b/new-metro.config.js
index abcdefg..hijklmn 100644
--- a/old-metro.config.js
+++ b/new-metro.config.js
@@ -1,17 +1,16 @@
-// Learn more https://docs.expo.io/guides/customizing-metro
 const { getDefaultConfig } = require('expo/metro-config');
+const path = require('path');

-/** @type {import('expo/metro-config').MetroConfig} */
-const config = getDefaultConfig(__dirname);
+// Find the project and workspace directories
+const projectRoot = __dirname;
+// This can be replaced with `find-yarn-workspace-root`
+const monorepoRoot = path.resolve(projectRoot, '../..');
+const config = getDefaultConfig(projectRoot);

-config.transformer.getTransformOptions = async () => ({
-  transform: {
-    // Inline requires are very useful for deferring loading of large dependencies/components.
-    // For example, we use it in app.tsx to conditionally load Reactotron.
-    // However, this comes with some gotchas.
-    // Read more here: https://reactnative.dev/docs/optimizing-javascript-loading
-    // And here: https://github.com/expo/expo/issues/27279#issuecomment-1971610698
-    inlineRequires: true,
-  },
-});
+// 1. Watch all files within the monorepo
+config.watchFolders = [monorepoRoot];
+// 2. Let Metro know where to resolve packages and in what order
+config.resolver.nodeModulesPaths = [
+  path.resolve(projectRoot, 'node_modules'),
+  path.resolve(monorepoRoot, 'node_modules'),
+];
 
-// This helps support certain popular third-party libraries
-// such as Firebase that use the extension cjs.
-config.resolver.sourceExts.push("cjs")
```

Speaking of which, this seems pretty drastic; are you sure you are using the latest Ignite here?

Reply:
Just a tip, we found on Cookbook that using `success-line` and `error-line` markup resulted in a better-looking diff with docusaurus:

https://github.com/infinitered/ignite-cookbook/blob/aca9fc9e6765ea8e5731ce375dc3231355558cfe/docs/recipes/EASUpdate.md?plain=1#L90-L95 for example

Reply:
That's very nice -- definitely cool!

![CleanShot 2024-08-13 at 16 13 39@2x](https://github.com/user-attachments/assets/41024374-a98b-47b0-b619-3ad06b560b40)